pkg/k8s/node.go

```diff
@@ -209,8 +217,25 @@ func ParseNode(k8sNode *types.Node, source source.Source) *node.Node {
 // GetNode returns the kubernetes nodeName's node information from the
 // kubernetes api server
 func GetNode(c kubernetes.Interface, nodeName string) (*v1.Node, error) {
-	// Try to retrieve node's cidr and addresses from k8s's configuration
-	return c.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
+	// Pass a context so that the call fails if it takes more than 120 seconds
+	ctx, cancel := context.WithTimeout(context.TODO(), 120*time.Second)
```

Comment:
https://github.com/cilium/cilium/blob/master/pkg/defaults/defaults.go is a good place to store timeout values